client/checkout/utils/upe.js

```diff
@@ -35,8 +35,9 @@ export const getSelectedUPEGatewayPaymentMethod = () => {
 	let selectedGatewayId = null;
 
 	// Handle payment method selection on the Checkout page or Add Payment Method page where class names differ.
-	const radio = document.querySelector(
-		'li.wc_payment_method input.input-radio:checked, li.woocommerce-PaymentMethod input.input-radio:checked'
+	const paymentContainer = document.querySelector( '#payment' );
+	const radio = paymentContainer?.querySelector(
+		'input[name="payment_method"]:checked'
```

Comment:
I considered switching from `input[name="payment_method"]` to just search for input of type `radio`, but after some thinking it seems that this type of implementation, while not relying on any IDs/classnames, might unexpectedly catch some unrelated radio inputs from within the payments container. Happy to hear your opinion on this in particular.